pyccel/parser/syntactic.py

```diff
@@ -657,6 +659,13 @@ def _visit_FunctionDef(self, stmt):
 
                 i = i+1
 
+            if (len(arguments) != args_number):
+                msg = 'The number of arguments in the function ({}) {} and the types decorator ({}) {} don\'t match.'.format(len(arguments), arguments, args_number, decorators['types'].args)
+                if (len(arguments) < args_number):
+                    errors.report(msg, severity='warning')
+                else:
+                    errors.report(msg, severity='error')
```

Comment:
Bounding box and symbol should be provided to help the user find the source of the problem